crypter.go

```diff
@@ -563,7 +579,9 @@ func (obj JSONWebEncryption) DecryptMulti(decryptionKey interface{}) (int, Heade
 		}
 	}
 
-	if plaintext == nil {
+	// BASE64URL(JWE Ciphertext) of an empty byte-string is an empty
+	// byte-string. If obj.ciphertext is empty, plaintext can be nil.
+	if plaintext == nil && len(obj.ciphertext) > 0 {
```

Comment:
You can have an empty plaintext with a non-empty ciphertext. Example:

```
{"protected":"eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2Q0JDLUhTNTEyIiwia2lkIjoibXlfa2V5In0","iv":"i8L9OP8YYfBkxLnQUblZ2g","ciphertext":"2kyoFmITGTY3MP7Ju1-3OQ","tag":"PrjWBte2feUSyUvwV5yf4qr9KXzwATz0gAfEuNnZiYo"}
```

That's produced by adding `fmt.Println(string(jwe.FullSerialize()))` to the original example from https://github.com/go-jose/go-jose/issues/43.

The "protected" header decodes to `{"alg":"dir","enc":"A256CBC-HS512","kid":"my_key"}`. The `ciphertext` field contains exactly 16 bytes, which is the block size of AES256CBC. That block is going to be all padding. Here's a [good answer to why there's an all-padding block](https://crypto.stackexchange.com/a/66649).

There's a TestRoundtripsJWE in crypter_test.go. Right now it always uses fixed input string (in the helper function RoundtripJWE). Probably the best way to test this change is to add a parameter "input" to the helper function, so TestRoundtripsJWE can test roundtrips of both empty and non-empty plaintexts. I believe that would catch the issue I'm talking about here.